psi4/src/psi4/libfock/jk.cc

```diff
@@ -540,6 +547,86 @@ void JK::AO2USO() {
     delete[] temp;
 }
 void JK::initialize() { preiterations(); }
+void JK::ifb_setup() {
+
+    if (iteration_ == 0 || D_ao_prev_.size() != D_ao_.size()) {
+
+        iteration_ = 0;
```

Comment:
I am _strongly_ opposed to the JK object needing to know the iteration number. I'll have a more concrete counter-recommendation when I can read the original paper.

Reply:
I'm still opposed to having an `iteration_` member variable, but as I look over the code, I don't understand it. Why are you treating D, J, K, and wK on the same footing in the `if` case, but treating D alone in the `else` case?